libraries/AP_WayBack/AP_WayBack.cpp

```diff
@@ -0,0 +1,818 @@
+/*
+
+this library collects points of drone position, records them in memory and eliminates 
+unnecessary data - redundant points and loops, so it always knows 
+the shortest way to home via visited points.
+ 
+ Copyright night_ghost@ykoctpa.ru 2017
+ 
+ license GPLV3 or above
+   
+*/
+
+#include "AP_WayBack.h"
+
+extern const AP_HAL::HAL& hal;
+
+AP_WayBack::Point* AP_WayBack::points;
+uint16_t          AP_WayBack::max_num_points=0; // size of storage
```

Comment:
You're indirectly using it like a singleton by making an assumption it will never be called twice.
Here's how DataFlash handles that check as an intentional singleton. For this type of class I suggest you simply make it like all the other classes so it's flexible.
https://github.com/ArduPilot/ardupilot/blob/master/libraries/DataFlash/DataFlash.h#L58